src/tests/ftest/control/dmg_storage_query.py

```diff
@@ -0,0 +1,198 @@
+        for dev in device_state_info:
+            if dev[2] != state:
+                self.fail("Found a device in {} state.".format(dev[2]))
+
+    def test_dmg_storage_query_smd_devices(self):
+        """
+        JIRA ID: DAOS-3925
+
+        Test Description: Test 'dmg storage query smd -d' command.
+
+        :avocado: tags=all,pr,hw,small,dmg_storage_query,smd_devs,basic
+        """
+        # Get the storage smd infromation, parse and check devices info
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", pools=False)
+        self.check_smd_out(smd_info, "devices", "pools")
+
+        # Check if the number of devices match the config
+        if self.check_len("dbev_list", len(smd_info)):
+            self.fail("Number of devices does not match config: {}".format(
+                len(smd_info)))
+
+    def test_dmg_storage_query_smd_pools(self):
+        """
+        JIRA ID: DAOS-3925
+
+        Test Description: Test 'dmg storage query smd -p' command.
+
+        :avocado: tags=all,pr,hw,small,dmg_storage_query,smd_pools,basic
+        """
+        # Create pool and get the storage smd information, then verfify info
+        self.prepare_pool(self.dmg)
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", devices=False)
+        self.check_smd_out(smd_info, "pools", "devices")
+
+        # Destroy pool and get the storage smd information, then verify info
+        self.pool.destroy()
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", devices=False)
+        self.check_smd_out(smd_info, "devices", "pools")
+
+    def test_dmg_storage_query_blobstore_health(self):
+        """
+        JIRA ID: DAOS-3925
+
+        Test Description: Test 'dmg storage query blobstore-health' command.
+
+        :avocado: tags=all,pr,hw,small,dmg_storage_query,blobstore,basic
+        """
+        # Create pool and get the storage smd information
+        self.prepare_pool(self.dmg)
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", devices=False)
+
+        # Parse stdout into a dict check that devices information is displayed.
+        blobstore_info = []
+        for info in smd_info:
+            if "Device" in info:
+                uuid = info[info.index("Device") + 1]
+                blobstore_info.append(
+                    self.get_dmg_cmd_info("storage_query_blobstore", uuid))
+
+        # Compare config expected values with dmg output
+        e_blob_info = self.params.get("blobstore_info", "/run/*")
+        if blobstore_info.sort() != e_blob_info.sort():
+            self.fail("dmg storage query expected output: {}".format(
+                e_blob_info))
+
+    def test_dmg_storage_query_device_state(self):
+        """
+        JIRA ID: DAOS-3925
+
+        Test Description: Test 'dmg storage query device-state' command.
+
+        :avocado: tags=all,pr,hw,small,dmg_storage_query,device_state,basic
+        """
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", devices=False)
```

Comment:
(pylint-unexpected-keyword-arg) Unexpected keyword argument 'devices' in method call
(pylint-no-value-for-parameter) No value for argument 'kwargs' in method call